src/main/java/com/synopsys/integration/detect/lifecycle/run/step/RlScanStepRunner.java

```diff
@@ -0,0 +1,151 @@
+import com.synopsys.integration.blackduck.service.request.BlackDuckResponseRequest;
+import com.synopsys.integration.blackduck.version.BlackDuckVersion;
+import com.synopsys.integration.detect.lifecycle.OperationException;
+import com.synopsys.integration.detect.lifecycle.run.data.BlackDuckRunData;
+import com.synopsys.integration.detect.lifecycle.run.operation.OperationRunner;
+import com.synopsys.integration.detect.util.bdio.protobuf.DetectProtobufBdioHeaderUtil;
+import com.synopsys.integration.detect.workflow.codelocation.CodeLocationNameManager;
+import com.synopsys.integration.exception.IntegrationException;
+import com.synopsys.integration.rest.HttpUrl;
+import com.synopsys.integration.rest.response.Response;
+import com.synopsys.integration.util.NameVersion;
+
+public class RlScanStepRunner {
+    
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+    private final NameVersion projectNameVersion;
+    private final String projectGroupName;
+    private UUID scanId;
+    private final OperationRunner operationRunner;
+    private final File rlRunDirectory;
+    private final BlackDuckRunData blackDuckRunData;
+    private String codeLocationName;
+    private static final String STORAGE_UPLOAD_ENDPOINT = "/api/storage/rldata/";
+    private static final String STORAGE_RL_CONTENT_TYPE = "application/vnd.blackducksoftware.rl-data-1+octet-stream";
+    private static final BlackDuckVersion MIN_BLACK_DUCK_VERSION = new BlackDuckVersion(2024, 4, 0);
+    
+    public RlScanStepRunner(OperationRunner operationRunner, BlackDuckRunData blackDuckRunData, NameVersion projectNameVersion) {
+        this.operationRunner = operationRunner;
+        this.blackDuckRunData = blackDuckRunData;
+        this.projectNameVersion = projectNameVersion;
+        
+        rlRunDirectory = operationRunner.getDirectoryManager().getReversingLabsOutputDirectory();
+        projectGroupName = operationRunner.calculateProjectGroupOptions().getProjectGroup();
+    }
+
+    public Optional<UUID> invokeRlWorkflow() {
+        try {
+            logger.debug("Determining if configuration is valid to run a ReversingLabs scan.");
+            if (!isReversingLabsEligible()) {
+                logger.info("No detect.rl.scan.file.path property was provided. Skipping ReversingLabs scan.");
+                return Optional.ofNullable(scanId);
+            }
+            if (!isBlackDuckVersionValid()) {
+                String minBlackDuckVersion = String.join(".", 
+                        Integer.toString(MIN_BLACK_DUCK_VERSION.getMajor()),
+                        Integer.toString(MIN_BLACK_DUCK_VERSION.getMinor()),
+                        Integer.toString(MIN_BLACK_DUCK_VERSION.getPatch()));
+                throw new IntegrationException("ReversingLabs scan is only supported with BlackDuck version "
+                        + minBlackDuckVersion + " or greater. ReversingLabs scan could not be run.");
+            }
+
```

Comment:
I think it might be good to also check if the RL scan file to upload was resolved from the property value provided before proceeding? I see there is a check to see if the `detect.rl.scan.file.path` was provided or not, but it is possible that the file path is invalid and the file wasn't found locally, so we can exit early -> refering to `isContainerImageResolved` in ContainerScanStepRunner. (Or is that checked someplace else?) 

In container scan, there's a download via URL feature as well, so checking if the file was fetched successfully was necessary (I guess URL download isn't an immediate requirement right now for RL but this may be applicable for local file retrieval)